### PR TITLE
feat: mask sensitive credentials in MCP responses and optionally save to .env

### PR DESCRIPTION
**Pull request by brth31**
## Summary

This PR introduces two security enhancements to the Auth0 MCP Server:

1. **Automatic credential masking** - Sensitive fields (e.g., `client_secret`, `signing_keys`, `private_key` etc.) are automatically redacted with `[REDACTED]` in all relevant MCP tool responses before being returned to AI clients. This prevents secrets from leaking into assistant logs or conversation history.

2. **Secure credential storage** - A new `auth0_save_credentials_to_file` tool allows users to securely persist Auth0 application credentials as environment variables to a local file. The AI assistant will automatically prompt users to save credentials after creating an application, or the tool can be invoked manually.

## Changes

- **Response masking**: Added a recursive masker that replaces sensitive field values (`client_secret`, `signing_keys`, `encryption_key`, `private_key`, etc.) with `[REDACTED]` in API response objects. Integrated into `auth0_get_application`, `auth0_create_application`, and `auth0_update_application` handlers.

- **Credential saving tool**: Added `auth0_save_credentials_to_file` tool that retrieves the full application credentials from Auth0 and writes them (`AUTH0_CLIENT_ID`, `AUTH0_CLIENT_SECRET`, `AUTH0_DOMAIN`, `AUTH0_CALLBACK_URL`) to a user-specified file path. If the file already exists, credentials are appended (preserving existing content); otherwise a new file is created with `chmod 600` permissions. The target file is automatically added to `.gitignore`.

- **Post-creation prompt**: After `auth0_create_application`, the response now includes a `_credentials_access` hint that instructs the AI assistant to ask the user for a file path before calling the save tool — no default path is assumed.

- **Documentation**: Added a "Credential Protection" subsection under the Security Best Practices section in `README.md`.

### Unit Tests

- **Response masker**: Covers primitives, nested objects, arrays, null/undefined values, custom sensitive field lists, case-insensitive matching, and edge cases.
- **Credentials writer**: Covers new file creation, appending to existing files, `.gitignore` creation and update, `chmod 600` permission setting, and error handling.
- **Application handlers**: Covers masked responses for `get`, `create`, `update` flows and the full `save_credentials_to_file` handler (including missing params, missing secret, 404, and 401 scenarios).

### Manual Tests

- Verified `auth0_create_application` masks secrets in the response and includes the `_credentials_access` prompt.
- Verified `auth0_save_credentials_to_file` creates a new `.env.local` with correct env vars and `600` permissions.
- Verified saving to an existing `.env` file appends credentials without overwriting prior content.
- Verified `.gitignore` is created or updated with the target file entry.
- Verified the AI assistant automatically prompts to save credentials after app creation.

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [ ] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [ ] All existing and new tests complete without errors

### CODE REVIEW

**Comment by brth31** [comment]
> @brth31 Adding a scenario related to tool composition. I’m wondering how this change will affect tool composition, especially when the output of one tool is used as input to another.
> 
> For example, a possible flow:
> 
> 1. Call `auth0_create_application` → returns a secret
> 2. Call `auth0_save_credentials_to_file` → writes the secret to `.env.local`
> 3. Use the Read tool to read `.env.local` and retrieve the value
> 4. Use that value as input for another tool call
> 
> A couple of concerns:
> 
> * If the MCP client uses the Read tool as part of chaining, wouldn’t that expose the secret in MCP client logs?
> * Reading `.env.local` may expose pre-existing secrets. If the file already contained sensitive values (secrets, API keys, etc.) before this tool appended to it, the mcp client's  read call would surface all of those in MCP client logs as
>   well, secrets this tool didn't create and the user didn't intend to expose.
> 
> Would this be a valid scenario ?

@gyaneshgouraw-okta these risks exist in any MCP environment where AI has file system access regardless of our tool. IMO our current approach balances security(masking secrets in responses) with DX(coding assistants can actually help with integration by writing secrets to .env). The read tool exposure is a client-side issue that affects all secret files not something we can solve at the server level.

File: src/tools/applications.ts [hunk 1]
```diff
@@ -256,6 +258,36 @@ export const APPLICATION_TOOLS: Tool[] = [
...
+      type: 'object',
+      properties: {
+        client_id: {
+          type: 'string',
+          description: 'Client ID of the application whose credentials should be saved',
+        },
+        file_path: {
+          type: 'string',
+          description:
+            'Required: Full or relative path to save credentials (e.g., ".env.local", "config/.env.development"). User must explicitly specify where to save credentials.',
```
[the remark this answers is not in the document]
Comment (gyaneshgouraw-okta) [reply src/tools/applications.ts]: Updated tool doc description in this commit to highlight behaviour if credential already exists.
https://github.com/auth0/auth0-mcp-server/pull/94/changes/6acc9459fb286136b7cc67098d86913a854dc9f9

Also updated the readme to include this information - https://github.com/auth0/auth0-mcp-server/pull/94/changes/621b43491f3174f5bbcc205c4bceff8724ee6c56

File: src/utils/response-masker.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+const DEFAULT_SENSITIVE_FIELDS = [
+  'client_secret',
+  'signing_keys',
+  'encryption_key',
+  'client_assertion',
+  'signing_key',
+  'secret',
+  'private_key',
+  'password',
+  'token',
```
[the remark this answers is not in the document]
Comment (gyaneshgouraw-okta) [reply src/utils/response-masker.ts]: Addressed this point in this commit - https://github.com/auth0/auth0-mcp-server/pull/94/changes/b062a48c488b4f6b9d74045f86b36730a674e32f
